File: trail1/별표 출력하기 12/print-star-12.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<vector<char>> board(N, vector<char>(N, ' '));
    for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
            if (i== 0) {board[i][j] = '*';}
            if (j % 2 == 1 && j >= i) {board[i][j] = '*';}
        }
    }
    for (int i=0; i<N; i++) {
        for (int j=0; j<N; j++) {
            cout << board[i][j] << ' ';
        }
        cout << "\n";
    }

    return 0;
}

File: trail1/별표 출력하기 12/print-star-12.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        char[][] board = new char[N][N];
        for (int i=0; i<N; i++) {
            for (int j=0; j<N; j++) {
                if (i==0 || (j % 2 == 1 && j >=i)) {board[i][j] = '*';}
                else {board[i][j] = ' ';} 
            }
        }
        for (int i=0; i<N; i++) {
            for (int j=0; j<N; j++) {
                System.out.print(board[i][j] + " ");
            }
            System.out.println();
        }
    }
}